### PR TITLE
chore: schema and types for log_pattern_policies

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

This is in a feature branch, ignore for serious review. Adds expconf for log pattern policies.

## Test Plan

Run this experiment
```
name: mnist_pytorch_const
data:
  url: https://s3-us-west-2.amazonaws.com/determined-ai-test-data/pytorch_mnist.tar.gz
hyperparameters:
  learning_rate: 1.0
  global_batch_size: 64
  n_filters1: 32
  n_filters2: 64
  dropout1: 0.25
  dropout2: 0.5
searcher:
  name: single
  metric: validation_loss
  max_length:
      batches: 937 #60,000 training images with batch size 64
  smaller_is_better: true
entrypoint: model_def:MNistTrial
log_pattern_policies:
  - pattern: "x"
    policy:
     type: on_failure_exclude_node
```

and see it doesn't fail

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by jgongd — APPROVED** [review]
LGTM
